sdk/testing/src/main/java/io/opentelemetry/sdk/testing/assertj/SpanDataAssert.java

```diff
@@ -159,6 +159,84 @@ public SpanDataAssert hasResource(Resource resource) {
     return this;
   }
 
+  /**
+   * Asserts the span has resource schemaUrl satisfying the given condition.
+   *
+   * @since 1.23.0
+   */
+  // Workaround "passing @Nullable parameter 'schemaUrl' where @NonNull is required", Nullaway
+  // seems to think assertThat is supposed to be passed NonNull even though we know that can't be
+  // true for assertions.
+  @SuppressWarnings("NullAway")
+  public SpanDataAssert hasResourceSchemaUrl(@Nullable String schemaUrl) {
+    isNotNull();
+    assertThat(actual.getResource().getSchemaUrl()).as("schemaUrl").isEqualTo(schemaUrl);
+    return this;
+  }
+
+  /**
+   * Asserts the span has resource attributes satisfying the given condition.
+   *
+   * @since 1.23.0
+   */
+  public SpanDataAssert hasResourceSatisfying(Consumer<Attributes> attributes) {
+    isNotNull();
+    assertThat(actual.getResource().getAttributes()).as("attributes").satisfies(attributes);
+    return this;
+  }
+
+  /**
+   * Asserts the event has resource attributes matching all {@code assertions}. Assertions can be
```

Comment:
Looks like some remnant typos from copy / paste. 

```suggestion
   * Asserts the span has resource attributes matching all {@code assertions}. Assertions can be
```